images/kindnetd/go.mod

```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/kind/images/kindnetd
 
-go 1.13
+go 1.18
```

Comment:
why don't you move to latest here then?

Reply:
it matched kubernetes at the time. note that this is not the go version used to compile, this is the language level features used.

Reply:
let's leave this for a later PR, this PR is already a lot and I don't want to deal with any breakage while trying to get the rest of the PR in.